src/app/data-passing.service.ts

```diff
@@ -0,0 +1,23 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
+
+export interface GroupCount {
+  numGroups: number;
+  numLinks: number;
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class DataPassingService {
```

Comment:
The class name indicate that it will be used for general purpose, but it only effect group count right now. What's the plan for this class in the future?

Reply:
Not really sure, I just figured I would probably need a data sharing service in the future. One thing that I was thinking though was that we could combine this one with the options service and the expand-collapse-all service since they all functionally do the same thing (just with different kinds of data). Also, while we are in the spirit of refactoring, we should probably rename link service to chrome-api service since that is a more accurate description of it's current responsibilities.

Reply:
I would rename this into group count service or stats service - in case we want to add other type of data. Then, combine option service with expand-collapse-all-service. I think it's reasonable to combine the one with similar theme.